File: src/main/java/com/wgzhao/addax/admin/config/ConfigInitializer.java
package com.wgzhao.addax.admin.config;

import com.wgzhao.addax.admin.service.SystemConfigService;
import lombok.AllArgsConstructor;
import org.springframework.boot.ApplicationArguments;
import org.springframework.boot.ApplicationRunner;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class ConfigInitializer implements ApplicationRunner
{
    private final SystemConfigService systemConfigService;

    @Override
    public void run(ApplicationArguments args)
    {
        systemConfigService.loadConfig();
        System.out.println("系统配置已加载完成");
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/config/ConfigInitializer.kt
package com.wgzhao.addax.admin.config

import com.wgzhao.addax.admin.service.SystemConfigService
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class ConfigInitializer(private val systemConfigService: SystemConfigService) {

    private val log = KotlinLogging.logger {}

    @EventListener
    fun onApplicationReady(@Suppress("UNUSED_PARAMETER") event: ApplicationReadyEvent) {
        systemConfigService.loadConfig()
        log.info { "系统配置已加载完成" }
    }
}
